engine_test.go

```diff
@@ -2414,23 +2438,35 @@ func mergableIndexDriver(tables map[string]*memory.Table) sql.IndexDriver {
 	})
 }
 
-func newUnmergableIndex(tables map[string]*memory.Table, tableName string, exprs ...sql.Expression) *memory.UnmergeableIndex {
-	return &memory.UnmergeableIndex{
-		DB:         "mydb",
-		DriverName: memory.IndexDriverId,
-		TableName:  tableName,
-		Tbl:        tables[tableName],
-		Exprs:      exprs,
+func newUnmergableIndex(tables map[string]sql.Table, tableName string, exprs ...sql.Expression) *memory.UnmergeableIndex {
```

Comment:
Actually can't do that, since the index implementation needs the memory type. But I'm actually going to revert these changes anyway, since they aren't really necessary (left over from the first draft)